homeassistant/components/recorder/history.py

```diff
@@ -919,7 +919,8 @@ def _sorted_states_to_dict(
         if row := initial_states.pop(ent_id, None):
             prev_state = row.state
             ent_results.append(state_class(row, attr_cache, start_time))
-            field_map = {key: idx for idx, key in enumerate(row._fields)}
+            if not field_map:
```

Comment:
```suggestion
            if field_map is None:
```

Reply:
Deleted this line per https://github.com/home-assistant/core/pull/88529#issuecomment-1438065409